pkg/utils/utils.go

```diff
@@ -28,3 +38,88 @@ func DeleteFinalizer(object metav1.Object, finalizer string) {
 	finalizers.Delete(finalizer)
 	object.SetFinalizers(finalizers.List())
 }
+
+// CheckClusterDeployment returns true if the ClusterDeployment is watched by this operator
+func CheckClusterDeployment(request reconcile.Request, client client.Client, reqLogger logr.Logger) (bool, *hivev1alpha1.ClusterDeployment, error) {
+
+	// remove SyncSetPostfix from name to lookup the ClusterDeployment
+	cdName := strings.Replace(request.NamespacedName.Name, config.SyncSetPostfix, "", 1)
+	cdNamespace := request.NamespacedName.Namespace
+
+	clusterDeployment := &hivev1alpha1.ClusterDeployment{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: cdName, Namespace: cdNamespace}, clusterDeployment)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			reqLogger.Info("No matching cluster deployment found, ignoring")
+			return false, clusterDeployment, nil
+		}
+		// Error finding the cluster deployment, requeue
+		return false, clusterDeployment, err
+	}
+
+	if clusterDeployment.DeletionTimestamp != nil {
+		return false, clusterDeployment, nil
+	}
+
+	if !clusterDeployment.Status.Installed {
+		return false, clusterDeployment, nil
+	}
+
+	if val, ok := clusterDeployment.GetLabels()[config.ClusterDeploymentManagedLabel]; ok {
+		if val != "true" {
+			reqLogger.Info("Is not a managed cluster")
+			return false, clusterDeployment, nil
+		}
+	} else {
+		// Managed tag is not present which implies it is not a managed cluster
+		reqLogger.Info("Is not a managed cluster")
+		return false, clusterDeployment, nil
+	}
+
+	// Return if alerts are disabled on the cluster
+	if _, ok := clusterDeployment.GetLabels()[config.ClusterDeploymentNoalertsLabel]; ok {
```

Comment:
If this label is set to "false", what do you expect to happen?

Reply:
It will still skip things, not checking the value of the label at all.